src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -338,6 +338,7 @@ internal sealed class MemberSignatureComparer : IEqualityComparer<Symbol>
             considerTypeConstraints: false,
             refKindCompareMode: RefKindCompareMode.ConsiderDifferences,
             considerCallingConvention: false,
+            considerArity: false,
```

Comment:
>false [](http://example.com/codeflow?start=27&length=5)

> although the legacy logic uses this comparer, it is not affected by this change since it only performs substitution/reduction once a unique signature is found.

Let's suppose we have two generic methods with different arity, but with the same signature. For example, none of them refer to their type parameters in the signature. Isn't this change going to affect their comparison before substitution?
It feels to me that keeping a separate comparer for the legacy logic is cheap enough price to pay for a piece of mind that the logic isn't broken.

Reply:
You're right, the difference is visible in that scenario. Added a test and a legacy comparer. Thanks!